cirq/ops/common_gates.py

```diff
@@ -81,6 +81,12 @@ def _apply_unitary_(self, args: protocols.ApplyUnitaryArgs
             args.available_buffer *= p
         return args.available_buffer
 
+    def as_rotation(self):
+        return XPowGate(exponent=self._exponent, global_shift=-0.5)
+
+    def as_pow_gate(self):
```

Comment:
Need better name. `XPowGate(...).as_pow_gate()` just looks wrong.

Maybe `with_global_phase_matching_smallest_period_cycle` or `as_pure_power_without_global_shift` or `with_no_phase_on_plus_one_eigenspace` or ... okay none of these are very good. But you get the idea.

Reply:
I didn't really envision the use as someone calling it in the way you described since it would only changed the gate if they specified a non-default `global_shift` in the constructor, in which case why would they specify that only to undo it. I agree that the name could use work nonetheless. I'm comfortable with relying on the fact that the user understands what `global_shift` parameter means since the only reason they would want to use these utility methods is to adjust it to their preferred gate specification. I can add docstrings to the methods to make their use more explicit as well.
How do the following sound?
`with_default_global_shift`
`with_default_global_phase`
`with_zero_global_shift`

Reply:
Because global shift is really not a concept I expect users to be familiar with, I want to downplay it. I really see the reason I would call this method as a way of canonicalizing the gate, where I decided to do so by moving all phase on to the - eigenbasis with no phase on the + eigenbasis (of the Pauli).

Perhaps what we need is a succinct way of referring to the space of single-qubit operations where one of the eigenvalues is equal to 1. Then we could just say `in_thatspace`. Maybe `with_determinant_equal_to_relative_phase`?

We could also go with something vague but user-relevant like `with_canonical_global_phase()`.

Reply:
I like `with_canonical_global_phase` because it implies a library default which is what this method provides.  I imagined these methods as translating between the "Cirq default" that you get if you didn't utilize `global_shift` at all and the "SU(2)" rotations. I expect the `in_su2` method to get much greater use by comparison. My expected use case for `with_canonical_global_phase` would be as a handy sanity check method for converting gates that you made with some `global_shift`, whether by directly specifying it in the constructor or by calling `in_su2` back into the "Cirq default". I'll make an update using those names.